packages/entity-server/src/app/addRoutesToApp.ts

```diff
@@ -5,15 +5,14 @@
 
 import { Express, Request, Response, NextFunction } from 'express';
 import { InternalServerError, RespondingError } from '@tupaia/utils';
-import { TestRoute } from '../routes';
+import { TestRoute, FetchEntityRoute } from '../routes';
 import { Route } from '../routes/Route';
 import { authenticationMiddleware } from '../auth';
+import { attachEntityToRequest, attachFormatterToResponse } from '../entity/middleware';
 
-const handleWith = (RouteClass: typeof Route) => (
-  req: Request,
-  res: Response,
-  next: NextFunction,
-) => {
+const handleWith = <Req extends Request, T extends Route<Req>>(
+  RouteClass: new (req: Req, res: Response, next: NextFunction) => T,
```

Comment:
Please don't be scared by this syntax, it means that the RouteClass will be a function that if you call `new` upon with the arguments provided then it will return an object of type `T`.

Speaking of type `T`, what are your thoughts @kael89 on using `T` here as the type alias (as opposed to something like `RouteType` or even `R`)?

Reply:
While I'm still not decided on the single vs multi letter generics dilemma, I think I'm leaning towards the single letter ones now (could be a matter of habit).

I remember I felt a bit confused when I was once reading `jest` types and they were using `E`, `T`, `R` generics. But after you get familiar with the context (`E` => expected, `R` => received etc), single letter generics probably make things simpler.

Overall, `T` seems like a good choice. Probably  better than `R` in this case, for the reason that both `Request` and `Response` start with `R`

Reply:
I'd go with `T`

Reply:
Yeah sorry I added that comment before you responded, but only just submitted the review as a whole - your thoughts above are great!